File: src/main/java/br/com/dusty/dkits/listener/mechanics/EntityDamageListener.java
package br.com.dusty.dkits.listener.mechanics;

import br.com.dusty.dkits.gamer.Gamer;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;

public class EntityDamageListener implements Listener {
	
	@EventHandler(priority = EventPriority.HIGHEST)
	public void onEntityDamage(EntityDamageEvent event) {
		if(event.getEntity() instanceof Player){
			Player player = (Player) event.getEntity();
			
			Gamer gamer = Gamer.of(player);
			
			if(gamer.isInvincible())
				event.setCancelled(true);
			
			if(event.getCause() == EntityDamageEvent.DamageCause.FALL && gamer.hasNoFall()){
				gamer.setNoFall(false);
				
				event.setCancelled(true);
			}
		}
	}
}

File: src/main/kotlin/br/com/dusty/dkits/listener/mechanics/EntityDamageListener.kt
package br.com.dusty.dkits.listener.mechanics

import br.com.dusty.dkits.util.gamer.gamer
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent

object EntityDamageListener: Listener {

	@EventHandler
	fun onEntityDamage(event: EntityDamageEvent) {
		if (event.entity is Player) {
			val player = event.entity as Player
			val gamer = player.gamer()

			if (gamer.warp.overrides(event)) return

			if (gamer.isInvincible) event.isCancelled = true

			if (event.cause == EntityDamageEvent.DamageCause.FALL && gamer.isNoFall) {
				gamer.isNoFall = false

				event.isCancelled = true
			}

			player.inventory.armorContents.forEach { it?.durability = 0 }
		}
	}
}
